browser/src/docstatefunctions.js

```diff
@@ -335,3 +335,23 @@ app.impress.isSlideSelected = function (index) {
 		return app.impress.partList[index].selected === 1;
 	} else return false;
 };
+
+app._flushLayoutingQueue = function () {
+	if (app.layoutTasks.length) {
+		window.requestAnimationFrame(() => {
+			const task = app.layoutTasks.shift();
+			if (task) {
+				task.call(this);
+				app.scheduleLayouting();
+			}
+		});
+	}
+};
```

Comment:
This does flush the queue, but only at a rate of 1 task/10ms... Is this intentional? It feels like either we'd want to service the entire queue here, or this should be called something else. Maybe `flushLayoutingQueueAsync` or `drainLayoutingQueue`.